torchvision/transforms/transforms.py

```diff
@@ -47,13 +51,30 @@ class Compose(object):
     """
 
     def __init__(self, transforms):
-        self.transforms = transforms
+        super().__init__()
 
-    def __call__(self, img):
+        try:
+            # Try to wrap the transforms that are not nn.Module into nn.Lambda
+            # register all transforms as nn.ModuleList <-> all transforms should be nn.Module
+            # if something fails we keep previous behaviour
+            new_transforms = []
+            for t in transforms:
+                if callable(t) and not isinstance(t, torch.nn.Module):
+                    t = Lambda(t)
```

Comment:
An issue here can be the following. For example, user would like to configure during the training some options of its custom transform and if it is wrapped by Lambda then previous code wont work...
```python

transforms = Compose([
    ...
    CustomTransform(some_config),  # at index 5
    ...
])

# assert hasattr(CustomTransform, "update_config")
# assert isinstance(transforms[5], Lambda)

for epoch in range(100):
     
    if epoch == 25:
        transforms[5].update_config(new_config)

```

As suggested by Philip, that we can "either enables attribute access to the underlying object or at least gives a meaningful error message".